flow-server/src/main/java/com/vaadin/flow/server/VaadinServlet.java

```diff
@@ -457,13 +457,13 @@ public VaadinServletService getService() {
      * @return false if cookies are disabled, true otherwise
      * @throws IOException
      */
-    private boolean ensureCookiesEnabled(VaadinServletRequest request,
-            VaadinServletResponse response) throws IOException {
+    private boolean ensureCookiesEnabled(VaadinRequest request,
+            VaadinResponse response) throws IOException {
         if (HandlerHelper.isRequestType(request, RequestType.UIDL)) {
             // In all other but the first UIDL request a cookie should be
             // returned by the browser.
             // This can be removed if cookieless mode (#3228) is supported
-            if (request.getRequestedSessionId() == null) {
+            if (request instanceof HttpServletRequest&&((HttpServletRequest)request).getRequestedSessionId() == null) {
```

Comment:
```suggestion
            if (request instanceof HttpServletRequest && ((HttpServletRequest) request).getRequestedSessionId() == null) {
```

@stbischof This could help with your rebuild as well ;) and makes the code a little bit easier to read.